File: Practica1/src/practica1/cliente.java
package practica1;
import java.util.Scanner;

public class cliente {
    
    public static void main(String[] args) {
        int numBicicletas;
        Scanner in;
        
        System.out.println("Elige el número de participantes para la carrera en carretera:");
        
        do {
            in = new Scanner(System.in);
        }while (!in.hasNextInt());
        
        numBicicletas = in.nextInt();
        
        FactoriaCarretera fcarretera = new FactoriaCarretera();
        FactoriaMontana fmontana = new FactoriaMontana();
        
        Carrera carreraC = fcarretera.crearCarrera(numBicicletas);
        carreraC.empezar();
        
        System.out.println("Elige el número de participantes para la carrera en montana:");
        
        do {
            in = new Scanner(System.in);
        }while (!in.hasNextInt());
        
        numBicicletas = in.nextInt();

        Carrera carreraM = fmontana.crearCarrera(numBicicletas);
        carreraM.empezar();
    }
    
}

File: Practica1Ruby/pkg/Practica1Ruby-0.0.1/lib/Cliente.rb
require_relative "FactoriaCarretera.rb"
require_relative "FactoriaMontana.rb"

public
class Cliente

	public
  
  def self.main()
    puts "Elige el numero de participantes para la carrera en carretera:"
        
    numBicicletas = gets.to_i
        
    fcarretera = FactoriaCarretera.new
    fmontana = FactoriaMontana.new
        
    carreraC = fcarretera.crearCarrera(numBicicletas)
    carreraC.empezar
        
    puts "Elige el numero de participantes para la carrera en montana:"
        
    numBicicletas = gets.to_i

    carreraM = fmontana.crearCarrera(numBicicletas)
    carreraM.empezar
    
    return(0)
  end
  
	def initialize()

	end
end

Cliente.main()

